muspinsim/__main__.py

```diff
@@ -24,29 +24,73 @@
 LOGFORMAT = "[%(levelname)s] [%(threadName)s] [%(asctime)s] %(message)s"
 
 
+def check_dir_path(string):
+    if os.path.isdir(string):
+        return string
+    else:
+        try:
+            os.makedirs(string)
+        except OSError:
+            raise NotADirectoryError(string)
+    return string
+
+
 def main(use_mpi=False):
 
     if use_mpi:
         mpi.connect()
 
     if mpi.is_root:
         # Entry point for script
-        parser = ap.ArgumentParser()
+        parser = ap.ArgumentParser(
+            description="Muspinsim - A program designed to carry out "
+            "spin dynamics calculations for muon science experiments"
+        )
         parser.add_argument(
-            "input_file",
+            "-o",
+            "--out-dir",
             type=str,
             default=None,
-            help="""YAML
-                            formatted file with input parameters.""",
+            help="""folder to store the output .dat files""",
+        )
+        parser.add_argument(
+            "-l",
+            "--log-path",
+            type=str,
+            default=None,
+            help="""filepath to store simulation logs """,
+        )
+        parser.add_argument(
+            "-f",
+            "--fitreport-path",
+            type=str,
+            default=None,
+            help="""filepath to store fit report if fitting parameters given""",
+        )
+        parser.add_argument(
+            "input_file",
+            type=ap.FileType("r"),
+            default=None,
+            help="""filepath to muspinsim specially formatted text
+            file specifying simulation input parameters.""",
         )
         args = parser.parse_args()
+        inp_filepath = args.input_file.name
 
-        fs = open(args.input_file)
+        fs = open(inp_filepath)
         infile = MuSpinInput(fs)
         is_fitting = len(infile.variables) > 0
 
         # Open logfile
-        logfile = "{0}.log".format(os.path.splitext(args.input_file)[0])
+        if args.log_path:
+            # check if directory exists, if not create it
+            logfile = "{0}/{1}".format(
+                check_dir_path(os.path.dirname(args.log_path)),
+                os.path.basename(args.log_path),
```

Comment:
We shouldn't assume that a user will always enter a path that includes directories. Using `-l test.log` raises `NotADirectoryError`